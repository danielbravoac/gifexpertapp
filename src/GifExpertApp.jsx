import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  };
  return (
    <div className="space-y-5">
      <h1 className="text-6xl text-center text-blue-500 font-bold">
        GifExpertApp
      </h1>
      <AddCategory onNewCategory={onAddCategory} />
      {categories.map((category) => (
        <GifGrid category={category} key={category} />
      ))}
    </div>
  );
};
