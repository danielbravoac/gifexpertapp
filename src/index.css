@tailwind base;
@tailwind components;
@tailwind utilities;

html {
  @apply p-9 bg-slate-950;
}
