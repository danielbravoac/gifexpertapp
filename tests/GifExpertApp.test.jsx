import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Pruebas en <GifExpertApp/>", () => {
  test("debe hacer match con el snapshot", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });
  test("debe de agregar nuevas secciones si se envia una nueva categoria", () => {
    render(<GifExpertApp />);
    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");
    fireEvent.input(input, { target: { value: "Iron Man" } });
    fireEvent.submit(form);
    //screen.debug();
    expect(screen.getByRole("heading", { level: 3 }).innerHTML).toBe(
      "Iron Man"
    );
  });
  test("should debe agregar la categoria si es Ãºnica", () => {
    render(<GifExpertApp />);
    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");
    fireEvent.input(input, { target: { value: "Iron Man" } });
    fireEvent.submit(form);

    fireEvent.input(input, { target: { value: "Iron Man" } });
    fireEvent.submit(form);
    expect(screen.getByRole("heading", { level: 3 }).innerHTML).toBe(
      "Iron Man"
    );
  });
});
