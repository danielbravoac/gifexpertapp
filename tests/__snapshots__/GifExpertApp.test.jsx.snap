// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="space-y-5"
  >
    <h1
      class="text-6xl text-center text-blue-500 font-bold"
    >
      GifExpertApp
    </h1>
    <form
      aria-label="form"
    >
      <input
        class="w-full p-2 border border-gray-300 rounded-md"
        placeholder="Buscar gifs"
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
